libs/sdk-core/src/breez_services.rs

```diff
@@ -2612,6 +2635,29 @@ async fn get_lsp_by_id(
         .cloned())
 }
 
+/// Convenience method to get all LSPs (active and historical) with whom we have an active channel
+async fn get_lsps_with_active_channel(
+    persister: Arc<SqliteStorage>,
+    lsp_api: Arc<dyn LspAPI>,
+    node_api: Arc<dyn NodeAPI>,
+) -> SdkResult<Vec<LspInformation>> {
+    let node_pubkey = persister
+        .get_node_state()?
+        .ok_or(SdkError::generic("Node info not found"))?
+        .id;
+
+    let used_lsps = lsp_api.list_used_lsps(node_pubkey).await?;
+
+    // Only consider LSPs with whom we have a channel
+    let mut lsps_with_active_channel = vec![];
+    for lsp in used_lsps {
```

Comment:
We only need to search for active channel in the historical lsps, not in the current lsp. We should register for the current lsp even if not channel exists since it will be used also to open a new channel during payment (we will need the notification in that case).

Reply:
Done: c9dfa4c06d4289ddb92a3c113d4c427ff0978da8